### ISSUE TITLE
[Rule change] Adding static/instance differentiation

### ISSUE THREAD

**Issue by FMorschel** [dart-code-checker/dart-code-metrics#859]
**What rule do you want to change?**
https://dartcodemetrics.dev/docs/rules/common/member-ordering-extended
https://dartcodemetrics.dev/docs/rules/common/member-ordering

**How will the change be implemented? (New option, new default behavior, etc.)?**
Adding:
- `static-public-fields`
- `static-private-fields`
- `static-public-getters`
- `static-private-getters`
- `static-public-setters`
- `static-private-setters`
- `static-getters-setters`
- `static-private-methods`
- `static-public-methods`
- `instance-public-fields`
- `instance-private-fields`
- `instance-public-getters`
- `instance-private-getters`
- `instance-public-setters`
- `instance-private-setters`
- `instance-getters-setters`
- `instance-private-methods`
- `instance-public-methods`

Or any better idea for differentiating them.

**Please provide some example code that this change will affect:** 

```yaml
dart_code_metrics:
  ...
  rules:
    ...
    -member-ordering:
        order:
          - constructors
          - static-public-fields
          - static-private-fields
          - instance-public-fields
          - instance-private-fields
```

**GOOD**
```dart
class A {
  static final bool staticPublicField = true;
  static final bool _staticPrivateField = true;
  final bool instancePublicField = true;
  final bool _instancePrivateField = true;
}
```

**BAD**
```dart
class A {
  final bool _instancePrivateField = true;
  // instance-public-fields should be before instance-private-fields // LINT
  final bool instancePublicField = true;
  // static-private-fields should be before instance-public-fields // LINT
  static final bool _staticPrivateField = true;
  // static-public-fields should be before static-private-fields // LINT
  static final bool staticPublicField = true;
}
```

**What does the rule currently do for this code?**
It doesn't differentiate ordering between static and instance methods/fields/getters/setters.

**What will the rule do after it's changed?**
It will differentiate ordering between static and instance methods/fields/getters/setters.

**Are you willing to submit a pull request to implement this change?**
I don't have the knowledge.

**Comment by incendial** [comment]
By parts I mean this 
`< (overridden | protected)- >< (private | public)- >< static- >< late- >< (var | final | const)- >< nullable- >< named- >< factory- > (fields | getters | getters-setters | setters | constructors | methods)`

**Comment by FMorschel** [comment]
I didn't see that before, thank you!
